hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/tokenkey/TokenKeyCall.java

```diff
@@ -69,10 +81,51 @@ public TokenKeyCall(
     private @NonNull FullResult fullResultsFor(
             @NonNull final ResponseCodeEnum status, final long gasRequirement, @NonNull final Key key) {
         // @Future remove to revert #9069 after modularization is completed
-        if (isStaticCall && status != SUCCESS) {
+        if ((isStaticCall && status != SUCCESS) || status == INVALID_TOKEN_ID || status == KEY_NOT_PROVIDED) {
```

Comment:
Why do we need to add INVALID_TOKEN_ID here? 
For non static calls to match mono behaviour in [call-results-snapshot.json](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/test-clients/src/main/resource/call-results-snapshot.json) Invalid token ID should result in top-level SUCCESS status with INVALID_TOKEN_ID without revert

Reply:
True but the suite tests expect the call to revert and since the task here is to adapt the logic so that the suites result in success. So basically the expected result from the suite test and the one from [call-results-snapshot.json](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/test-clients/src/main/resource/call-results-snapshot.json) contradict each other

Reply:
I tested with this branch locally and removed the INVALID_TOKEN_ID from this if check and the test are still passing.

From what I saw from the suite it seems like this check that requires REVERT with INVALID_TOKEN_ID is for Update_TXN and not for tokenKey. Could you double-check, please?

Reply:
Yes, they do indeed. I've removed the mentioned checks. Good catch!